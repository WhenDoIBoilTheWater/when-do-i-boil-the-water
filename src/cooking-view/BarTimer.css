:root{
    --color-1: black;
}

.bar-timer-bar{
    animation-name: 'barTimerAnimation';
    animation-iteration-count: infinite;
    background-color: var(--color-1);
    height: 5px;
}

.bar-timer-field{
    background-color: grey;
    width: 100wv;
}

@keyframes barTimerAnimation {
    from {
        width: 0vw;
    }

    to {
        width: 100vw;
    }
}

