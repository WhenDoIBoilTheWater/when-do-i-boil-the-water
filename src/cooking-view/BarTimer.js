import React from 'react'
import './BarTimer.css'

class BarTimer extends React.Component {
    constructor(props){
        super(props)

    }

    render(){
        let length = `${this.props.length}s`
        console.log(length)
        const styles = {
            // animationDuration: '5s'
            animationDuration: length,
            width: '0%'
        }

        return(
            <div className="bar-timer-field">
                <div style={styles} className="bar-timer-bar" />
            </div>
        )
    }
}

export default BarTimer