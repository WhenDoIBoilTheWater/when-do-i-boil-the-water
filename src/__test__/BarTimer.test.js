import React from 'react';
import BarTimer from '../cooking-view/BarTimer.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BarTimer />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
