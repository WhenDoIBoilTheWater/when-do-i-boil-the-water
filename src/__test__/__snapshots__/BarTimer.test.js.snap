// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bar-timer-field"
>
  <div
    className="bar-timer-bar"
    style={
      Object {
        "animationDuration": "undefineds",
      }
    }
  />
</div>
`;
